### PR TITLE
Limit interpreter depth in precompute, but not when running whole modules

### PR DESCRIPTION
**Pull request by kripken**
Keep limiting in precompute as before: that is useful since that pass is run as part of normal compilation, and we want to avoid native stack limits on all platforms. Also that pass is not likely to find any pattern of size 50 of higher that it can't precompute as a sum of smaller pieces.

Restore the 250 limit from before for interpreting entire modules, as without that the fuzzer will sometimes hit the limit and cause a false positive.

### CODE REVIEW

File: src/wasm-interpreter.h [hunk 1]
```diff
@@ -129,13 +127,16 @@ class Indenter {
 template<typename SubType>
 class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
 protected:
-  // Keep a record of call depth, to guard against excessive recursion.
-  size_t depth = 0;
+  Index maxDepth;
+
```
Comment (tlively) [inline src/wasm-interpreter.h]: nit: this whitespace looks unnecessary.

File: src/passes/Precompute.cpp [hunk 2]
```diff
@@ -41,6 +41,12 @@ namespace wasm {
 
 static const Name NOTPRECOMPUTABLE_FLOW("Binaryen|notprecomputable");
 
+// Limit evaluation depth for 2 reasons: first, it is highly unlikely
+// that we can do anything useful to precompute a hugely nested expression
+// (we should succed at smaller parts of it first). Second, a low limit is
+// helpful to avoid platform differences in native stack sizes.
+enum { MAX_DEPTH = 50 };
```
Comment (tlively) [inline src/passes/Precompute.cpp]: Why use an enum here?

File: src/wasm-interpreter.h (hunk 1 above)
Comment (kripken) [reply src/wasm-interpreter.h]: It's sort of separating the input values from the internal state values. Is there a clearer way to do that? I could use another `protected:` block lower down I guess.

File: src/passes/Precompute.cpp (hunk 2 above)
Comment (kripken) [reply src/passes/Precompute.cpp]: It could be a `static const` or a `#define` I guess. I don't have a preference, what do you think?

Comment (tlively) [reply src/passes/Precompute.cpp]: I would default to using a `static const`. That way the underlying type is clear and there are no extra type shenanigans happening.

File: src/wasm-interpreter.h (hunk 1 above)
Comment (tlively) [reply src/wasm-interpreter.h]: Nope, this newline seems fine if it's there intentionally :+1:

**Pull request merged**